File: backend/src/main/java/com/ecommerce/backend/service/OrderService.java
package com.ecommerce.backend.service;

import com.ecommerce.backend.dto.OrderDTO;
import com.ecommerce.backend.dto.view.OrderDetailView;
import com.ecommerce.backend.dto.view.OrderStatistics;
import com.ecommerce.backend.dto.view.OrderView;
import com.ecommerce.backend.model.Order;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface OrderService {
     Page<OrderView> getAllOrders(Pageable pageable);
     OrderDetailView getOrderById(long id);
     Order createOrder(OrderDTO orderDTO);
     Order updateOrder(long id, OrderDTO orderDTO);
     void deleteOrder(long id);
     OrderStatistics orderStatistics();
     Page<OrderView> getAllOrdersByStatus(String status, Pageable pageable);
}

File: frontend_client_mobile/lib/services/order_service.dart
import 'package:frontend_client_mobile/models/PageResponse.dart';
import 'package:frontend_client_mobile/models/order_statistics.dart';
import 'package:frontend_client_mobile/models/order_view.dart';
import 'package:frontend_client_mobile/services/api/api_client.dart';
import 'package:frontend_client_mobile/services/api/order_api_service.dart';

class OrderService {
  final OrderApiService _apiOrderService = ApiClient.getOrderApiService();

  Future<PageResponse<OrderView>> getOrders({
    String? status,
    String? sortBy = "createdDate",
    String? direction = "DESC",
    int page = 0,
    int size = 10,
  }) async {
    final response = await _apiOrderService.getOrders(
      status,
      sortBy!,
      direction!,
      page,
      size,
    );
    return response.data;
  }

  Future<OrderView> createOrder(OrderView order) async {
    return await _apiOrderService.createOrder(order);
  }

  Future<OrderView> createOrderFromCart({
    required List<int> cartItemIds,
    String? voucherCode,
  }) async {
    return await _apiOrderService.createOrderFromCart({
      'cartItemIds': cartItemIds,
      'voucherCode': voucherCode,
    });
  }

  Future<OrderView> updateOrder(int id, OrderView order) async {
    return await _apiOrderService.updateOrder(id, order);
  }

  Future<void> deleteOrder(int id) async {
    return await _apiOrderService.deleteOrder(id);
  }

  Future<OrderView> getOrder(int id) async {
    return await _apiOrderService.getOrder(id);
  }

  Future<OrderStatistics> getStatistics() async {
    return await _apiOrderService.getStatistics();
  }
}
